src/vm/svml-machine.ts

```diff
@@ -584,8 +585,59 @@ function INTERNAL_FUNCTION_CALL() {
   }
 }
 
-function SET_TO() {
-  TO = Math.ceil((0.5 + Math.random() * 0.5) * TO_MAX)
+let scheduler: Scheduler = new RoundRobinScheduler()
+const threads: Map<ThreadId, any> = new Map()
```

Comment:
hm is it possible to define a tuple type to represent a thread? or perhaps just a object